IRSA-663:TimeSeries tool: Make it easier for users to find the right entry in the Background Monitor

lznakano opened the pull request:
Changed the downloaded file name to meaningful name
  Fixed the bug in DownloadDialog.jsx
  Adjust the BackgroundMonitor.css's BGMon's size
  Removed the reduanant codes in time series
  Adjust the fields in the download dialog size
  Add a labelWidth property in DownloadDialog.jsx

To test:
1.  Check out this branch and build firefly
2. There is no IFE changes, use ife in dev is fine and build irsaviewer
3. Run time series

ejoliet commented on src/firefly/js/templates/lightcurve/ptf/PTFMissionOptions.js:
```diff
@@ -149,17 +149,19 @@ export function ptfOnFieldUpdate(fieldKey, value) {
     return (
         <DownloadButton>
             <DownloadOptionPanel
+                groupKe = {mission}
```
'groupKey' is misspelled. I'm not sure is used but you probably want to fix the name.

lznakano replied:
It is a typo.  I will fix it.

ejoliet commented on src/firefly/js/templates/lightcurve/ptf/PTFMissionOptions.js:
```diff
@@ -149,17 +149,19 @@ export function ptfOnFieldUpdate(fieldKey, value) {
  * @param cutoutSizeInDeg
  * @returns {XML}
  */
 export function ptfDownloaderOptPanel (mission, cutoutSizeInDeg) {
```
Why is ptf having it's own download panel? Can't it share the default one like WISE?

loitly replied:
`FileGroupProcessor` is different.  Beside that, I don't see any differences.  It would not be hard to define a `download` property to LcConverterFactory reserved for download parameters.  But, it does not have to be done this way.

ejoliet replied:
i will raise a separate ticket to cleanup the code.

ejoliet commented on src/firefly/js/templates/lightcurve/LcResult.jsx:
```diff
@@ -143,12 +155,13 @@ const StandardView = ({visToolbar, title, searchDesc, imagePlot, xyPlot, tables,
                         initialState={{
-                            value: 'A sample download',
-                            label: 'Title for this download:'
+                            value: `${m}_Files:${currentTime}`,
```
In order to make the label entry shorter and useful, i would suggest to format the current time to a more minimalist version: MM/DD/YY - HH:mm:ss (Local time). No need of the day label, and no need of the time zone mention either. 
Example today: 02/20/18 17:36:27
JavaScript toLocaleString() can help for that.

lznakano replied:
I will make the time stamp shorter

lznakano commented:
The reason that PTF has its own download panel is that it has different search parameters, such as processor, productLevel, table etc.

lznakano commented:
Changes made as requested.  @loitly  I made some changes in DownloadDialog.jsx.  Please take a look at it.   The I removed the hard coded groupKey in DownloadPanel.  I added a labelWidth property.  

Thanks!

ejoliet commented:
But the download panel accept 'mission' so it can know what to do. I don't think we want another panel. We will have the same problem for ZTF in near term. Can we use the same default panel for PTF the same way we do for WISE? You have to show me.

loitly commented on src/firefly/js/core/background/BackgroundMonitor.css:
```diff
@@ -7,8 +7,8 @@
-    width: 330px;
-    min-width: 330px;
+    width: 500px;
+    min-width: 500px;
```
Changing this setting will affect the original layout.  It was designed to support single, double, and triple columns display.  Notice the max-width of 590px below?  Changing min to 500 would limit the range from 500-590 which does not make much sense anymore.  Before I look into how this setting affect the outcome old the original use-case, can I ask why it was necessary to change it in the first place?

loitly commented on src/firefly/js/ui/DownloadDialog.jsx:
```diff
@@ -141,9 +143,9 @@ export class DownloadOptionPanel extends SimpleComponent {
+        const {groupKey, cutoutSize, help_id, children, style, title, tbl_id, dlParams, dataTag, labelWidth} = this.props;
         const {mask, email, enableEmail} = this.state;
-        const labelWidth = 110;
+        const lblWidth = labelWidth|| 110;
```
No need to define this awkward `lblWidth` variable.  Just use `labelWidth` and default it to 110 at its declaration line.. i.e.
`...  dataTag, labelWidth=110} = this.props;`

lznakano commented:
Hi Emmanuel,


I DID NOT add  PTFDownloadPanel.  It is an existing issue.  If you want 
to remove it, it is fine.  I can do so.  Do you want to submit a new 
ticket for the changes?

For this ticket, it asked to give meaningful names for the download 
files.  I implemented as what it was asked and meanwhile I fixed a bug 
found that stopped from showing the changes I made.


On 2/21/18 2:19 PM, Emmanuel Joliet wrote:
>
> But the download panel accept 'mission' so it can know what to do. I 
> don't think we want another panel. We will have the same problem for 
> ZTF in near term. Can we use the same default panel for PTF the same 
> way we do for WISE? You have to show me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/552#issuecomment-367495273>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ALbTfV2a4G5a4JV-e5rOtookZdXSwDRvks5tXJZcgaJpZM4SM4Ho>.
>

-- 
Lijun Zhang

LSST/IPAC
California Institute of Technology
770 S. Wilson Ave, Mail Code 100-22
Pasadena, CA 91125, USA
Voice: 1-626-395-1972 Fax:1-626-395-1816

lznakano commented:
- I reverted the BackgroundMonitor.css as it was in the dev.   I removed the added labelWith property in DownloadDialog.jsx,  I cut the label name short with the "Title For.. Download" so that the default length "110" is long enough.

- The issue asked about PtfMissionOption.jsx is out of the scope.  I was not the original author and did not know the reason whatever.   If a change is needed, please SUBMIT a new ticket.  

- Please test the most recent commit.

Thanks!
